bin/qunit

```diff
@@ -0,0 +1,768 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+require "optparse"
+require "shellwords"
+require "json"
+require "fileutils"
+require "socket"
+require "uri"
+
+class QunitRunner
+  def initialize(args)
+    @file_paths = []
+    @filter = nil
+    @verbose = false
+    @dry_run = false
+
+    @ember_port = ENV["EMBER_PORT"]&.to_i || 4200
+    @rails_port = ENV["UNICORN_PORT"]&.to_i || 3000
+
+    @full_mode = false
+    @parallel = ENV["QUNIT_PARALLEL"]
+    @seed = ENV["QUNIT_SEED"]
+    @target = ENV["TARGET"]
+    @module = ENV["MODULE"]
+    @qunit_skip_core = ENV["QUNIT_SKIP_CORE"]
+    @qunit_single_plugin = ENV["QUNIT_SINGLE_PLUGIN"]
+    @theme_name = ENV["THEME_NAME"]
+    @theme_url = ENV["THEME_URL"]
+    @theme_id = ENV["THEME_ID"]
+    @qunit_path = nil
+    @rails_env = ENV["QUNIT_RAILS_ENV"] || "test"
+    @report_requests = ENV["REPORT_REQUESTS"] == "1"
+    @load_plugins = ENV["LOAD_PLUGINS"] == "1"
+
+    parse_options(args)
+  end
+
+  def run
+    if @dry_run
+      dry_run_tests
+    elsif @full_mode
+      run_full_mode
+    elsif @file_paths.any?
+      run_file_tests
+    elsif @filter
+      run_filter_tests
+    else
+      run_all_tests_quick
+    end
+  end
+
+  private
+
+  def show_help
+    puts <<~HELP
+      Usage: bin/qunit [options] [files|directories]
+
+      Runs the Discourse QUnit suite through Ember Exam. With no arguments it runs
+      the entire suite against your dev servers; add --full to spin up an isolated
+      Rails/Ember stack (the same flow CI uses).
+
+      Options
+        -f, --filter TEXT          Filter tests by QUnit module/test name
```

Comment:
We're currently duplicating all of the params & documentation in `#show_help` and in `#parse_options`.

OptionParser has a built-in system for generating help, including the ability to add a custom banner. Can we use that? ([docs](https://ruby-doc.org/stdlib-2.7.1/libdoc/optparse/rdoc/OptionParser.html#class-OptionParser-label-Generating+Help))